### PR TITLE
Added convenience functions for Uniforms

### LINKED ISSUE #5: Renderer : Create convenience functions to access / modify uniforms
Create some convenience function in the Shader class to easily manage uniforms in the used shaders. Should probably look a little like this:
`void setUniform{type of uniform} (const std::string name, {type of uniform} value);`

### PR DESCRIPTION
**Pull request by SapphireNick**
Done all functions for int and float vectors.

Closes #5

### CODE REVIEW

**Comment by SapphireNick** [comment]
I realized in #20 the math-module, you @d-jakob wrapped the functions in the source file in namespaces again yet i didn't. Both work and both would work without them. How should we do it?

File: modules/renderer/src/r_shader.cc [hunk 1]
```diff
@@ -81,3 +85,35 @@ Shadowheart::Renderer::Shader::Shader(const std::string& vertexShaderPath, const
...
+  glUniform2i(glGetUniformLocation(_id, name_of_uniform.c_str()), data[0], data[1]);
+}
+
+void Shadowheart::Renderer::Shader::setUniform3i(std::string name_of_uniform, Math::ivec3 data) {
+  glUniform3i(glGetUniformLocation(_id, name_of_uniform.c_str()), data[0], data[1], data[2]);
+}
+
+void Shadowheart::Renderer::Shader::setUniform4i(std::string name_of_uniform, Math::ivec4 data) {
+  glUniform4i(glGetUniformLocation(_id, name_of_uniform.c_str()), data[0], data[1], data[2], data[3]);
+}
```
Comment (d-jakob) [inline modules/renderer/src/r_shader.cc:119]: I thought we are using 
```
rtype fname(params)
{
  code
}
```
?

**Review by d-jakob — CHANGES_REQUESTED** [review]
I think wrapping them in the namespaces again improves readability as it will make it possible to remove the `Shadowheart::Renderer::` from every function

**Comment by SapphireNick** [comment]
> I think wrapping them in the namespaces again improves readability as it will make it possible to remove the `Shadowheart::Renderer::` from every function

Noted. Will change this.

File: modules/renderer/src/r_shader.cc (hunk 1 above)
Comment (SapphireNick) [reply modules/renderer/src/r_shader.cc:119]: Well the newlines do not show, oh well..

**Review by d-jakob — APPROVED** [review]
I added the lf at eof myself this time

**Comment by SapphireNick** [comment]
Thank you !

**Pull request merged**